torch/csrc/distributed/c10d/Backend.hpp

```diff
@@ -386,6 +386,12 @@ class TORCH_API Backend : public torch::CustomClassHolder {
     // backends may perform
   }
 
+  // whether the backend is fully initialized, e.g., for NCCL, if the NCCL comms
+  // are fully initialized and ready to use.
+  virtual bool isInitialized() {
+    return false;
+  }
+
```

Comment:
Could it be a bit unsafe to assume a default value here?
Maybe we should throw an unimplemented error here and force backends to implement it? But it will induce more work, including contacting 3rd-party backends to implement this.

Reply:
If this is problemm We could move this interface to nccl backend only, similar to other APIs such as bound_device_id or abort